micro-rdk/src/common/registry.rs

```diff
@@ -212,138 +230,150 @@ impl ComponentRegistry {
         Ok(())
     }
 
-    pub fn register_encoder(
+    pub fn register_encoder<S: Into<String>>(
         &mut self,
-        model: &'static str,
+        model: S,
         constructor: &'static EncoderConstructor,
     ) -> Result<(), RegistryError> {
-        if self.encoders.contains_key(model) {
+        let model = model.into();
+        if self.encoders.contains_key(&model) {
             return Err(RegistryError::ModelAlreadyRegistered(model));
         }
         let _ = self.encoders.insert(model, constructor);
         Ok(())
     }
 
-    pub fn register_base(
+    pub fn register_base<S: Into<String>>(
         &mut self,
-        model: &'static str,
+        model: S,
         constructor: &'static BaseConstructor,
     ) -> Result<(), RegistryError> {
-        if self.bases.contains_key(model) {
+        let model = model.into();
+        if self.bases.contains_key(&model) {
             return Err(RegistryError::ModelAlreadyRegistered(model));
         }
         let _ = self.bases.insert(model, constructor);
         Ok(())
     }
 
-    pub fn register_power_sensor(
+    pub fn register_power_sensor<S: Into<String>>(
         &mut self,
-        model: &'static str,
+        model: S,
         constructor: &'static PowerSensorConstructor,
     ) -> Result<(), RegistryError> {
-        if self.power_sensors.contains_key(model) {
+        let model = model.into();
+        if self.power_sensors.contains_key(&model) {
             return Err(RegistryError::ModelAlreadyRegistered(model));
         }
         let _ = self.power_sensors.insert(model, constructor);
         Ok(())
     }
 
-    pub fn register_servo(
+    pub fn register_servo<S: Into<String>>(
         &mut self,
-        model: &'static str,
+        model: S,
         constructor: &'static ServoConstructor,
     ) -> Result<(), RegistryError> {
-        if self.servos.contains_key(model) {
+        let model = model.into();
+        if self.servos.contains_key(&model) {
             return Err(RegistryError::ModelAlreadyRegistered(model));
         }
         let _ = self.servos.insert(model, constructor);
         Ok(())
     }
 
-    pub fn register_generic_component(
+    pub fn register_generic_component<S: Into<String>>(
         &mut self,
-        model: &'static str,
+        model: S,
         constructor: &'static GenericComponentConstructor,
     ) -> Result<(), RegistryError> {
-        if self.generic_components.contains_key(model) {
+        let model = model.into();
+        if self.generic_components.contains_key(&model) {
             return Err(RegistryError::ModelAlreadyRegistered(model));
         }
         let _ = self.generic_components.insert(model, constructor);
         Ok(())
     }
 
-    pub fn register_dependency_getter(
+    pub fn register_dependency_getter<S: Into<String>>(
         &mut self,
-        component_type: &'static str,
-        model: &'static str,
+        component_type: S,
+        model: S,
```

Comment:
This is why I think using `impl Into<String>` on the args would be a little better, because as written, `component_type` and `model` must be the same type.